components/faction/commons/faction.lua

```diff
@@ -35,28 +36,44 @@ Faction.types.FactionProps = TypeUtil.struct{
 	faction = Faction.types.Faction,
 }
 
-local byName = Table.map(Data.factionProps, function(faction, props) return props.name, faction end)
-local byLowerName = Table.map(byName, function(name, faction) return name:lower(), faction end)
+---@generic U, T
+---@param func fun(key?: string, value?: table): U, T
+---@return fun(key?: string, value?: table): U, T
+local transformWrapper = function(func)
+	if Faction.byGame then
+		return function (game, factionProps)
+			return game, Table.map(factionProps, func)
+		end
+	end
+	return func
+end
+
+local byName = Table.map(Data.factionProps, transformWrapper(function(faction, props) return props.name, faction end))
+local byLowerName = Table.map(byName, transformWrapper(function(name, faction) return name:lower(), faction end))
```

Comment:
should we maybe `FnUtil.memoize` it?

Reply:
I don't think that is really necessary for two invocations, or am i misunderstanding?